hyper-parameter-optimization.ipynb

```diff
@@ -0,0 +1,180 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get resources"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import coiled\n",
+    "import dask.distributed"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cluster = coiled.Cluster(\n",
+    "    n_workers=20, \n",
+    "    configuration=\"coiled/default\", \n",
+    ")\n",
+    "client = dask.distributed.Client(cluster)\n",
+    "\n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get and pre-process data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask.dataframe as dd\n",
+    "df = dd.read_csv(\n",
+    "    \"s3://nyc-tlc/trip data/yellow_tripdata_2019-*.csv\", \n",
+    "    parse_dates=[\"tpep_pickup_datetime\", \"tpep_dropoff_datetime\"],\n",
+    "    dtype={\n",
+    "        \"VendorID\": \"UInt8\",\n",
+    "        \"passenger_count\": \"UInt8\",\n",
+    "        \"RatecodeID\": \"UInt8\",\n",
+    "        \"store_and_fwd_flag\": \"category\",\n",
+    "        \"PULocationID\": \"UInt16\",\n",
+    "        \"DOLocationID\": \"UInt16\",    \n",
+    "        \"payment_type\": \"UInt8\",\n",
```

Comment:
Nit: Some columns are included here are never seen again, like `PULocationID`.

Reply:
Yeah, this was a copy-paste job from another notebook.  I should probably remove some of these columns with `usecols=` I guess.